src/injecting/injector.rs

```diff
@@ -72,20 +93,92 @@ impl FunctionMapper for Module {
             })
             .ok_or("No malloc function in table section")?;
 
+        Ok(malloc_index)
+    }
+    fn map_function(
+        &mut self,
+        function_name: &str,
+        body_mapper: impl Fn(&mut FuncBody, usize),
+    ) -> Result<(), String> {
+        // NOTE:
+        // Indexing for local functions includes both imported and own (local)
+        // functions. So we actually need the imported function count to resolve the
+        // actual index of the given function in own functions list.
+
+        // Find the global function index
+        let global_function_index = self.get_global_function_index(function_name).unwrap();
+
+        // Count the number of imported functions listed by the module
+        let import_section_len: usize = self.get_import_section_len().unwrap();
+
+        let malloc_index = self.get_malloc_index().unwrap();
+
+        // Compute the local function index (for the code section) by subtracting
+        // the number of imported functions from the global function index
+        let local_function_index = global_function_index - import_section_len;
         // Extract the `${function_name}` function_body
         let function_body = self
-            .code_section_mut()
-            .ok_or("No code section")?
-            .bodies_mut()
-            .get_mut(local_function_index)
-            .ok_or(format!(
-                "Function '{}' not found in the code section",
-                function_name
-            ))?;
+            .get_function_body(local_function_index, function_name)
+            .unwrap();
 
         // Map over the function_body
         body_mapper(function_body, malloc_index);
 
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::util::load_module_from_wasm;
+    use std::path::PathBuf;
+
+    // NOTE VALUES ARE FOR TEST WASM ONLY AND WILL DIFFER FOR DIFFERENT WASM BLOBS!!!
+    const VALIDATE_BLOCK_GLOBAL_INDEX: usize = 1733;
+    const IMPORT_SECTION_LENGTH: usize = 39;
+    const MALLOC_INDEX: usize = 25;
+    const WASM_INSTRUCTION_COUNT: usize = 358;
+
+    #[test]
+    fn test_get_global_function_index() {
+        let mut d = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        d.push("test.wasm");
```

Comment:
Might want to extract this `PathBuf` (or even the loaded module for that matter) as a constant. You can test the module loading in isolation.